tools/const_finder/lib/const_finder.dart

```diff
@@ -72,12 +74,42 @@ class _ConstVisitor extends RecursiveVisitor<void> {
     super.visitConstructorInvocation(node);
   }
 
+  @override
+  void visitClass(Class node) {
+    // check if this is a class that we should ignore
+    if (_isStaticIconProvider(node)) {
+      inIgnoredClass = true;
+      super.visitClass(node);
+      inIgnoredClass = false;
+      return;
+    }
+    // not an ignored class
+    super.visitClass(node);
+  }
+
+  // Constants from classes annotated with an instance of StaticIconProvider
+  // should be ignored.
+  bool _isStaticIconProvider(Class node) {
+    return node.annotations.any((Expression expression) {
+      if (expression is! ConstantExpression) {
+        return false;
+      }
+
+      final DartType type = expression.type;
+      if (type is InterfaceType && type.classNode.name == 'StaticIconProvider') {
+        return true;
+      }
+      return false;
```

Comment:
Can you do this? Or does type promotion not work for this?

```suggestion
    return expression is! ConstantExpression
      && expression.type is InterfaceType
      && expression.type.classNode.name == 'StaticIconProvider';
```

Reply:
You need to make `expression.type` a local, the compiler can't promote fields. I can clean this up a little though...

Reply:
got rid of three lines--I think this is as compact as I can make this without casts.